docs/general/data-structures.md

```diff
@@ -0,0 +1,39 @@
+# Data structures
+
+The `data` property of a dataset can be passed in various formats. By default all data is parsed using configured scales.
+
+## Primitive[]
+
+```javascript
+data: [20, 10],
+labels: ['a', 'b']
+```
+
+When the `data` is an array of numbers, values from `labels` array at the same index are used for the index axis (x for vertical, y for horizontal charts).
+
+## Object[]
+
+```javascript
+data: [{x: 10, y: 20}, {x: 15, y: 10}]
+```
+
+```javascript
+data: [{x:'2016-12-25', y:20}, {x:'2016-12-26', y:10}]
+```
+
+```javascript
+data: [{x:'Sales', y:20}, {x:'Revenue', y:10}]
+```
+
+This is also the internal format used for parsed data. Property names are matched to scale-id. In this mode, parsing can be skipped by specifying `parsed: true` in the dataset. If parsing is skipped, data must be in the formats the configured scales accept internally.
```

Comment:
It would be good to add a note about this to `performance.md` as well

Reply:
added a line and link.